test/Microsoft.AspNetCore.Razor.Test.Common/Langauge/IntegrationTests/IntegrationTestBase.cs

```diff
@@ -147,26 +148,48 @@ protected void AssertCSharpDocumentMatchesBaseline(RazorCSharpDocument document)
             }
 
             var baselineFilename = Path.ChangeExtension(Filename, ".codegen.cs");
+            var baselineDiagnosticsFilename = Path.ChangeExtension(Filename, ".diagnostics.txt");
 
             if (GenerateBaselines)
             {
                 var baselineFullPath = Path.Combine(TestProjectRoot, baselineFilename);
                 File.WriteAllText(baselineFullPath, document.GeneratedCode);
+
+                var baselineDiagnosticsFullPath = Path.Combine(TestProjectRoot, baselineDiagnosticsFilename);
+                var lines = document.Diagnostics.Select(RazorDiagnosticSerializer.Serialize).ToArray();
+                if (lines.Any())
+                {
+                    File.WriteAllLines(baselineDiagnosticsFullPath, lines);
+                }
+                else if (File.Exists(baselineDiagnosticsFullPath))
+                {
+                    File.Delete(baselineDiagnosticsFullPath);
+                }
+
                 return;
             }
 
-            var testFile = TestFile.Create(baselineFilename, GetType().GetTypeInfo().Assembly);
-            if (!testFile.Exists())
+            var codegenFile = TestFile.Create(baselineFilename, GetType().GetTypeInfo().Assembly);
+            if (!codegenFile.Exists())
             {
                 throw new XunitException($"The resource {baselineFilename} was not found.");
             }
 
-            var baseline = testFile.ReadAllText();
+            var baseline = codegenFile.ReadAllText();
 
             // Normalize newlines to match those in the baseline.
             var actual = document.GeneratedCode.Replace("\r", "").Replace("\n", "\r\n");
-
             Assert.Equal(baseline, actual);
+
+            var baselineDiagnostics = string.Empty;
+            var diagnosticsFile = TestFile.Create(baselineDiagnosticsFilename, GetType().GetTypeInfo().Assembly);
+            if (diagnosticsFile.Exists())
+            {
+                baselineDiagnostics = diagnosticsFile.ReadAllText();
+            }
+
+            var actualDiagnostics = string.Concat(document.Diagnostics.Select(d => RazorDiagnosticSerializer.Serialize(d) + "\r\n"));
```

Comment:
Because we normalize the baselines when we read them. I forbid you from ever commenting on this again :trollface: